client/allocrunner/networking_bridge_linux.go

```diff
@@ -156,69 +165,61 @@ func (b *bridgeNetworkConfigurator) Teardown(ctx context.Context, alloc *structs
 	return b.cni.Teardown(ctx, alloc, spec)
 }
 
-func buildNomadBridgeNetConfig(b bridgeNetworkConfigurator, withConsulCNI bool) []byte {
-	var consulCNI string
-	if withConsulCNI {
-		consulCNI = consulCNIBlock
+func buildNomadBridgeNetConfig(b bridgeNetworkConfigurator, withConsulCNI bool) ([]byte, error) {
+	// Update website/content/docs/networking/cni.mdx when the bridge configuration
+	// is modified. If CNI plugins are added or versions need to be updated for new
+	// fields, add a new constraint to nomad/job_endpoint_hooks.go
+	var ranges []Range
+	allocRange := Range{Subnet: b.allocSubnet}
+	ranges = append(ranges, allocRange)
+
+	var routes []Route
+	allocRoute := Route{Dst: "0.0.0.0/0"}
+	routes = append(routes, allocRoute)
+
+	loop := Loopback{Type: "loopback"}
+
+	NewFirewall := Firewall{
+		Type:                   "firewall",
+		Backend:                "iptables",
+		IptablesAdminChainName: cniAdminChainName,
+	}
+	NewPortmap := Portmap{
+		Type:         "portmap",
+		Capabilities: CapabilityArgs{Portmappings: true},
+		Snat:         true,
+	}
+
+	NewBridge := &Bridge{
+		Type:         "bridge",
+		Bridgename:   b.bridgeName,
+		IpMasq:       true,
+		IsGateway:    true,
+		ForceAddress: true,
+		HairpinMode:  b.hairpinMode,
+		Ipam: IPAM{
+			Type:   "host-local",
+			Routes: routes,
+		},
 	}
+	NewBridge.Ipam.Ranges = append(NewBridge.Ipam.Ranges, ranges)
```

Comment:
is this split out like this because over in ipv6-land we'll have more than this one range?

Reply:
Yep! That's the idea 👍

Reply:
this confused me for a minute, because I have tried to append a slice to a slice like this before, and was mistaken in those cases, but here `Ranges` is a `[][]Range`!  so it's appending a slice to a slice of slices.

but! I think I would personally set the default ipv4 `Ranges` up in the main `&Bridge{` definition, then later we can append the ipv6 range(s) to it in a separate line like this -- akin to how you're doing below `withConsulCNI`

Reply:
That makes more sense to me as well, we can do that!